modules/stitching/src/matchers.cpp

```diff
@@ -583,7 +583,10 @@ void SurfFeaturesFinderGpu::find(InputArray image, ImageFeatures &features)
     image_.upload(image);
 
     ensureSizeIsEnough(image.size(), CV_8UC1, gray_image_);
-    cvtColor(image_, gray_image_, COLOR_BGR2GRAY);
+
+    Mat gray_image;
+    cvtColor(image, gray_image, COLOR_BGR2GRAY);
+    gray_image_.upload(gray_image);
```

Comment:
Thinking more about it, this just does not feels right. @sadeqbillah this approach might work, but this is wrong. Check this logic:

```
// Colored image uploaded
<cv::cuda::gpuMat> image_.upload(upload)
// converted grayscale on the gpu itself.
cv::cuda::cvtColor(image_, gray_image_, COLOR_BGR2GRAY)
// we do not need to convert to grayscale and upload again.
```

Agreed the problem is here, but this is not the right solution.